api/lib/opentelemetry/metrics/meter.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Metrics
+    # No-op implementation of Meter.
+    module Meter
+      extend self
+
+      NOOP_DOUBLE_MEASURE = DoubleMeasure.new
+      NOOP_LONG_MEASURE = LongMeasure.new
+
+      private_constant(:NOOP_DOUBLE_MEASURE, :NOOP_LONG_MEASURE)
+
+      def record(*measurements, distributed_context: nil, exemplar: nil)
+        raise ArgumentError if measurements.empty?
+      end
+
+      def create_measure(name, description: nil, unit: nil, type: :double)
+        raise ArgumentError if name.nil? # TODO: The Java implementation also constrains the name to be printable and 255 chars or less.
+
+        case type
+        when :double
+          NOOP_DOUBLE_MEASURE
+        when :long
+          NOOP_LONG_MEASURE
+        else
+          raise ArgumentError
+        end
+      end
+
+      # TODO: I contemplated using '..., **constant_labels)' here instead of the optional named arg, but that allocates an empty hash if no additional args are passed.
+      def create_double_gauge(name, description: nil, unit: nil, component: nil, resource: nil, label_keys: nil, constant_labels: nil)
+        raise ArgumentError if name.nil?
+        raise ArgumentError if label_keys&.any?(nil)
+        raise ArgumentError if constant_labels&.any? { |k, v| k.nil? || v.nil? }
+
+        DoubleGauge.new(label_keys_size: label_keys&.size || 0)
+      end
+
+      # TODO: I'm not sure how I feel about these "typed" method names. Would it be preferable to pass 'type:' as an argument?
```

Comment:
I don't expect the list of types will grow significantly in the future. So I don't see a strong argument to change it. It's also plausible that if types are added in the future, they could be specialized enough to require additional type-dependent arguments.

Reply:
That said, I just realized that `create_measure` uses a `type` argument. Maybe we should prioritize consistency here.